### PR TITLE
shell: windows: make EnvMap case-insensitive

### PR DESCRIPTION
**Pull request by nektro**
breaking out https://github.com/oven-sh/bun/pull/9578 into multiple PRs in order to better pin down where my CI is going awry

### CODE REVIEW

File: src/shell/interpreter.zig [hunk 1]
```diff
@@ -509,11 +509,17 @@ pub const EnvMap = struct {
     const MapType = std.ArrayHashMap(EnvStr, EnvStr, struct {
         pub fn hash(self: @This(), s: EnvStr) u32 {
             _ = self;
+            if (bun.Environment.isWindows) {
```
Comment (jdalton) [inline src/shell/interpreter.zig:512]:
```suggestion
            if (comptime bun.Environment.isWindows) {
```
comptimes?

File: src/shell/interpreter.zig [hunk 2]
```diff
@@ -509,11 +509,17 @@ pub const EnvMap = struct {
...
+                return bun.CaseInsensitiveASCIIStringContext.hash(undefined, s.slice());
+            }
             return std.array_hash_map.hashString(s.slice());
         }
         pub fn eql(self: @This(), a: EnvStr, b: EnvStr, b_index: usize) bool {
             _ = self;
             _ = b_index;
+            if (bun.Environment.isWindows) {
+                return bun.CaseInsensitiveASCIIStringContext.eql(undefined, a.slice(), b.slice(), undefined);
+            }
```
Comment (jdalton) [inline src/shell/interpreter.zig:522]: I believe the internals of zig do this kind of thing for their env map logic for windows too

File: src/shell/interpreter.zig (hunk 1 above)
Comment (nektro) [reply src/shell/interpreter.zig:512]: already is

**Review by gvilums — APPROVED** [review]

**Comment by Jarred-Sumner** [comment]
nice

**Pull request merged**